server.js

```diff
@@ -1,34 +1,92 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
+import listEndpoints from 'express-list-endpoints'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import netflixData from './data/netflix-titles.json'
+
+// Defines the port the app will run on. Defaults to 8080
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// Middlewares 
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
+// Route to return a list of the endpoints available
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Query routes to search for title, director and/or release year
+app.get('/titles', (req, res) => {
+  const { title, director, releaseyear } = req.query
+  let titles = netflixData
+
+  if (title) {
+    titles = titles
+      .filter((item) => item.title.toString().toLowerCase().replace(/\s+/g, '') === title.toLowerCase().replace(/\s+/g, ''))
+  }
+  if (director) {
+    titles = titles
+      .filter((item) => item.director.toString().toLowerCase().replace(/\s+/g, '') === director.toLowerCase().replace(/\s+/g, ''))
+  }
+  if (releaseyear) {
+    titles = titles
+      .filter((item) => item.release_year === +releaseyear)
+  }
+
+  res.json({ length: titles.length, data: titles })
+})
+
+// Route to return a single movie/tvshow object, when request is an exact Url
+app.get('/titles/:title', (req, res) => {
+  const { title } = req.params
+  const findShow = netflixData
+    .find((show) => show.title.toString().toLowerCase().replace(/\s+/g, '') === title.toLowerCase().replace(/\s+/g, ''))
+  
+  if (findShow) {
+    res.status(200).json({ data: findShow })
+  } else {
+    res.status(400).json({ error: 'Title not found, check your spelling or try another title' })
```

Comment:
You using this structure a lot, so it is a great candidate to extract it to the separate function 🤓